File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    private static readonly string[] ResistorColors = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color) => Array.IndexOf(ResistorColors, color);

    public static string[] Colors() => ResistorColors;
}

File: nim/resistor-color/resistor_color.nim
import sequtils

type
  ResistorColor* = enum
    Black, Brown, Red, Orange, Yellow,
    Green, Blue, Violet, Grey, White

proc colorCode*(color: ResistorColor): int = color.ord

proc colors*(): seq[ResistorColor] = toSeq(ResistorColor.items)
